lib/decompress.ps1

```diff
@@ -98,6 +125,58 @@ function Expand-7zipArchive {
             abort "Failed to list files in $Path.`nNot a 7-Zip supported archive file."
         }
     }
+    if ($IsTarZst) {
+        # always remove extracted .tar file
+        Remove-Item $Path -Force
+        $Path = $OrigPath
+    }
+    if ($Removal) {
+        # Remove original archive file
+        Remove-Item $Path -Force
+    }
+}
+
+function Expand-ZstdArchive {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [String]
+        $Path,
+        [Parameter(Position = 1)]
+        [String]
+        $DestinationPath = (Split-Path $Path),
+        [String]
+        $ExtractDir,
+        [Parameter(ValueFromRemainingArguments = $true)]
+        [String]
+        $Switches,
+        [ValidateSet("All", "Skip")]
+        [String]
+        $Overwrite,
+        [Switch]
+        $Removal
+    )
+
+    $ZstdPath = Get-HelperPath -Helper Zstd
+    $LogPath = "$(Split-Path $Path)\zstd.log"
+    $ArgList = @('-d', "-o `"$DestinationPath`"", "`"$Path`"")
+
+    if ($Switches) {
+        $ArgList += (-split $Switches)
+    }
+    switch ($Overwrite) {
+        "All" { $ArgList += "-f" }
+        "Skip" { $ArgList += "-q" }
+    }
+
+    $Status = Invoke-ExternalCommand $ZstdPath $ArgList -LogPath $LogPath
```

Comment:
Use try catch and abort in case of `[System.Management.Automation.ParameterBindingException]`

https://github.com/Ash258/Scoop-Core/blob/96f1407439d2199a244d67b7a02cdf6d7cb36cf6/lib/decompress.ps1#L89-L93

Reply:
Ok, will do that